### PR TITLE
Latin american countries

### PR DESCRIPTION
**Pull request by glatimer**
- [x] Created Router
- [x] Created list of Countries
- [x] Created Population bar chart
- [x] Created list of Languages (WIP)

### CODE REVIEW

File: src/components/fetchData.jsx
Comment (NAlexH2) [inline src/components/fetchData.jsx:1]: Ya'll are really doin it right with the separate function for the fetch! I kept my file count much smaller and this just makes it easier to read. Good job! I also had a different way of parsing all the data too, which is messy. Learning from you all on what thoughtful code vs rushed code (me lol) has been a great experience.

File: src/index.css
Comment (NAlexH2) [inline src/index.css:1]: Good css! It makes the rest much easier to read too, though I see some bootstrap which is great too! You did a great job of blending the two together.

File: src/routes/language-page.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <div>
+      <p>Loading...</p>
+    </div>;
+  }
+
+  // map through the api making an array of official_languages, no duplicates
+  const duplicates = countries.map((country) => country.official_languages);
+  const languages = countries
+    .map((country) => country.Languages)
+    .filter(...duplicates);
```
Comment (NAlexH2) [inline src/routes/language-page.jsx:16]: Can you explain to me how you're using the spread operator here? This sort of syntax still eludes me in my understanding of it.

File: src/utils/chartColors.js
Comment (NAlexH2) [inline src/utils/chartColors.js:1]: Good idea to export them all to a separate thing like this. 

I would recommend using a hashes, or associative arrays (called a dictionary in python) for future lists of data that need to be accessed specifically at a location. It has O(1) access and you could avoid long load times too I think if you only access the data when you need it vs. loading it with the page.

**Review by NAlexH2 — COMMENTED** [review]
Great work. I really like your CSS.

File: src/components/population-chart.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { Bar } from "react-chartjs-2";
+import useFetchData from "./fetchData";
+import { backgroundColors, borderColors } from "../utils/chartColors";
+
+export default function BarChart() {
+  const { countries, loading } = useFetchData();
+
+  if (loading) {
```
Comment (PhatCash) [inline src/components/population-chart.jsx:9]: Great use of showcasing the loading returned from the fetch. I did something similar, but I instead rendered a component. Both ways work, but I found it helpful to render a component.

File: src/routes/list-page.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        <h1>Countries of Latin America</h1>
+        <Root />
+        <p>
+          This route should show a list of the countries of Latin America, their
+          flags, and information about each country.
+        </p>
+
+        <main className="container">
+          <div className="grid">
+            {countries.map((country) => (
```
Comment (PhatCash) [inline src/routes/list-page.jsx:25]: I would suggest checking if the data exists. This was mentioned in class where some APIs have missing data, so we must check and provide alternatives in the form of error checking.

File: src/components/population-chart.jsx
Comment (baylie-r-g) [inline src/components/population-chart.jsx:1]: Running your program in codespace, it appears that the charts aren't loading. I vaguely remember Caterina going over the issue of the "Canvas already in use" error.

Comment (baylie-r-g) [reply src/components/population-chart.jsx:1]: image

Comment (baylie-r-g) [reply src/components/population-chart.jsx:1]: This problem is present in both charts

**Review by baylie-r-g — COMMENTED** [review]
The home and list page look great! I like the consistency of cards. The charts aren't running on my machine but I am not sure if that is just me. Styling wise, I would have centered the navbar and put some margin below to create some white space. Great router!

File: src/index.css
Comment (glatimer) [reply src/index.css:1]: Yes! I tried to take advantage of bootstrap's responsive abilities :)

File: src/routes/language-page.jsx (hunk 1 above)
Comment (glatimer) [reply src/routes/language-page.jsx:16]: lol it doesn't work.  I haven't gotten back to it yet.  I think that I'm misusing the spread op here

File: src/utils/chartColors.js
Comment (glatimer) [reply src/utils/chartColors.js:1]: Great idea!

File: src/components/population-chart.jsx
Comment (glatimer) [reply src/components/population-chart.jsx:1]: Dang, thanks!  I will look into it.  The bar chart should be working, but the pie chart is still WIP

**Comment by PhatCash** [comment]
Looks like I never submitted a review. Submitting now!

**Pull request merged**